### ISSUE TITLE
Publish an npm package to assist npm builds

### ISSUE THREAD

**Issue by junderw** [rust-lang/rustup#3401]
### Problem you are trying to solve

In a NodeJS project, we are compiling some rust modules that use NAPI-RS.

Currently we have a note in the INSTALL.md that says "hey install rust here, do that, do this, and THEN run `npm ci` to install the packages from the lockfile"

We would like to make this process simpler.

### Solution you'd like

If we could add a `"rustup": "x.x.x",` dependency to our nodejs project, then `npm ci` would just automatically pull down rustup. It would also automatically pull down @napi/cli, and as a part of the install script just start building the rust from source.

### Notes

We would like to avoid including pre-built binaries of our NAPI-RS binaries inside our project.

Also, I noticed that someone has already had the same idea. https://www.npmjs.com/package/rustup

We would rather use a package maintained by someone well known and respected in the NodeJS and/or Rust communities.

**Comment by rbtcollins** [comment]
What does 'official package' mean here?

I would be happy to collaborate with someone on the npm side who wants to maintain whatever it is you're looking for, but I'm skeptical of the `rustup` team itself taking on the maintenance of such a thing, partly because there are many language ecosystems that might want that (C, C++, Python, NPM, Java, ...) and partly because we're already stretched.

As far as pre-built binaries go, I think you're out of luck, since to build `rustup` or `rustc` you have to have a working rust toolchain, and the way you get a working rust toolchain is by starting with a working binary.

**Comment by junderw** [comment]
> As far as pre-built binaries go, I think you're out of luck

Sorry, what I meant was we don't want to include our NAPI binary in our project as a pre-built.

The whole point of this issue is that we would like pre-built rustup binaries pulled down as a part of the npm install process so that people without rust won't need to do anything other than what they normally do.

The current rustup package on npm works fine, and I have verified the source is basically just copying what the rustup sh file does... But some random unknown account is the maintainer...

It would be nice if someone well known and respected published it.

For now I can just pin the exact version since I know npm doesn't allow re-publishing the same exact version, and I verified it myself.
